libraries/AC_WPNav/AC_Circle.cpp

```diff
@@ -306,26 +306,29 @@ void AC_Circle::get_closest_point_on_circle_NEU_m(Vector3p& result_neu_m, float&
 
     // Compute vector from stopping point to the circle center
     Vector3f vec_to_center_neu_m = (stopping_point_neu_m - _center_neu_m).tofloat();
-    dist_m = vec_to_center_neu_m.length();
-
     // Return circle center if radius is zero (vehicle orbits in place)
     if (!is_positive(_radius_m)) {
         result_neu_m = _center_neu_m;
+        dist_to_edge_m = 0;
         return;
     }
 
+    const float dist_to_center_m_sq = vec_to_center_neu_m.length_squared();
     // Handle edge case: vehicle is at the exact center of the circle
-    if (is_zero(dist_m)) {
+    if (is_zero(dist_to_center_m_sq)) {
```

Comment:
It's another existing issue but I think we should actually loosen the condition so that if the vehicle is within some small number (perhaps 10% of the radius or 1m) we use this logic.  This would resolve the somewhat random direction that the vehicle takes off in when entering circle mode

This is of course not a blocker for this PR